Tests/BowLiteCoreTests/Data/State/State+Arbitrary.swift

```diff
@@ -2,14 +2,24 @@ import SwiftCheck
 import BowLiteLaws
 import BowLiteCore
 
-extension State: Arbitrary where StateType: Hashable & Arbitrary & CoArbitrary, Value: Arbitrary {
-    
+extension State: Arbitrary where StateType: Hashable & CoArbitrary & Arbitrary, Value: Arbitrary {
     public static var arbitrary: Gen<State<StateType, Value>> {
-        ArrowOf<StateType, Pair<StateType, Value>>.arbitrary.map { arrow in
+        ArrowOf<StateType, Tuple<StateType, Value>>.arbitrary.map { arrow in
             State { state in
                 let pair = arrow.getArrow(state)
                 return (pair.first, pair.second)
             }
         }
     }
 }
+
+private struct Tuple<A, B> {
+    let first: A
+    let second: B
+}
+
+extension Tuple: Arbitrary where A: Arbitrary, B: Arbitrary {
+    static var arbitrary: Gen<Tuple<A, B>> {
+        Gen.zip(A.arbitrary, B.arbitrary).map(Tuple.init)
+    }
+}
```

Comment:
Why do you need to repeat this, if it is already included in `Pair`?

Reply:
`Pair` is more complex than a simple `tuple` and this generator is not working properly in Linux. Using `ArrowOf<S, Pair>` for testing in Linux, throw the next error during the compile-time (while emitting swift intermediate language):

```bash
2.	While emitting IR SIL function "@$s11BowLiteCore5StateV0abC5TestsSHRz10SwiftCheck9ArbitraryRzAE02CoH0RzAeFR_rlE9arbitraryAE3GenVyACyxq_GGvgZAkE7ArrowOfVyxAA4PairVyxq_GAqefDxAeFHD2__q_AeFHD4_HCg1_GcfU_".
 for expression at [/mnt/hgfs/47deg/bow-lite/Tests/BowLiteCoreTests/Data/State/State+Arbitrary.swift:8:66 - line:13:9] RangeText="{ arrow in
            State { state in
                let pair = arrow.getArrow(state)
                return (pair.first, pair.second)
            }
        "
3.	While mangling type for debugger type 'ArrowOf<StateType, Pair<StateType, Value>>'

```